modules/http/http.h

```diff
@@ -30,10 +30,13 @@
 
 #include "logthrdestdrv.h"
 
+#define CURL_NO_OLDIES 1
```

Comment:
Just a note to other reviewers how I tried to validate this: the problem was introduced in libcurl with 9adf3c473a01b289c781aab111f9ad2fc541ed4e (libcurl 7.50), which was fixed in d6604524ad24daf4581efbe0020da058d2b3af84 (libcurl 7.51) by introducing the underlying define.

So as I understand, this has effect only in very recent libcurl  >= 7.51 (for comparison, xenial ships with 7.47), and has no effect below <7.50. In summary, if this compiles with xenial or trusty, then probably will on all supported platforms.